drivers/hmis/app/models/hmis/hud/processors/geolocation_processor.rb

```diff
@@ -13,51 +13,81 @@ def schema
       nil
     end
 
-    def process(field, value)
-      return @processor.send(factory_name).destroy if value.nil? || value.empty?
+    def relation_name
+      :clh_location
+    end
 
+    def process(field, value)
       attribute_name = ar_attribute_name(field)
-
+      # The only expected attribute name is 'coordinates'
       raise ArgumentError, "Unexpected attribute for Geolocation: #{attribute_name}" unless attribute_name == 'coordinates'
 
-      attribute_value = JSON.parse(value, symbolize_names: true)
+      # if 'Geolocation.coordinates: nil' was submitted, destroy the clh location record
+      return destroy_record if value.nil? || value.empty?
+
+      attribute_value = clean_coordinate_value(value)
       latitude = attribute_value[:latitude]
       longitude = attribute_value[:longitude]
       not_collected_reason = attribute_value[:notCollectedReason]
 
-      raise ArgumentError, 'Geolocation coordinates in unexpected format' unless (latitude && longitude) || not_collected_reason
-
-      return @processor.send(factory_name).destroy if not_collected_reason
+      return destroy_record if not_collected_reason # not collected reason is returned from external PIT form
+      return destroy_record if latitude.nil? && longitude.nil?
```

Comment:
what if we get just one coord (not likely). But maybe it should early-return unless we have both
```suggestion
      return destroy_record unless latitude && longitude
```